src/librustc/hir/lowering/expr.rs

```diff
@@ -659,11 +659,23 @@ impl LoweringContext<'_> {
         // mut pinned => loop { ... }
         let pinned_arm = self.arm(pinned_pat, loop_expr);
 
+
+        // `match ::std::future::IntoFuture::into_future(<expr) { ... }`
+        let expr = P(self.lower_expr(expr));
+        let into_future_expr = {
+            let into_future_path =
+                &[sym::future, sym::IntoFuture, sym::into_future];
```

Comment:
This `let` binding feels rather redundant; let's inline it into where it is used (and inline the tail expression into the `let into_future_expr`).

Reply:
I copied the pattern from the `IntoIterator` code ;)

Reply:
Figured, let's fix the `IntoIterator` code also then.

Reply:
Updated (only here, left `IntoIterator` alone in this PR).